lib/ansible/modules/cloud/vagrant/vagrant_box.py

```diff
@@ -0,0 +1,281 @@
+            opts_dict['--box-version'] = self.version
+
+        # conditional options, they are added ONLY IF the box is a filename
+        if os.path.exists(self.box):
+            if self.name is not None:
+                opts_dict['--name'] = self.name
+            if self.checksum is not None:
+                opts_dict['--checksum'] = self.checksum
+            opts_dict['--checksum-type'] = self.checksum_type
+
+        if opts_dict:
+            for k, v in opts_dict.items():
+                opts_list.append(str(k) + " " + str(v))
+
+        cmd = "vagrant box add %s %s" % (self.box, " ".join(opts_list))
+        rc, out, err = self.module.run_command(cmd)
+
+        if rc == 0:
+            self.module.exit_json(changed=True, rc=rc, stdout=out, stderr=err)
+        elif "already exists" in err:
+            self.module.exit_json(changed=False, stdout="The box %s version: %s for provider: %s already exists" % (self.box, self.version, self.provider))
+        else:
+            self.module.fail_json(rc=rc, stdout=out, stderr=err)
+
+    def delete(self):
+        if not self.ispresent():
+            self.module.exit_json(changed=False, stdout="The box %s version: %s for provider: %s already removed" % (self.box, self.version, self.provider))
+
+        opts_list = []
+        if self.force:
+            opts_list.append('--force')
+        if self.provider:
+            opts_list.append('--provider %s' % self.provider)
+        if self.version is not "latest":
+            opts_list.append('--box-version "%s"' % self.version)
+
+        cmd = "vagrant box remove %s %s" % (self.box, " ".join(opts_list))
+        rc, out, err = self.module.run_command(cmd)
+
+        if rc == 0:
+            self.module.exit_json(changed=True, rc=rc, stdout=out, stderr=err)
+        elif "interface with the UI" in err:
+            self.module.fail_json(msg="The box %s is still in use by at least one Vagrant environment, use 'force' flag to force the deletion" % self.box)
+        else:
+            self.module.fail_json(msg=err)
+
+    def update(self):
+        if not self.ispresent() and self.version is not "latest":
+            self.module.fail_json(msg="The box %s version: %s for provider: %s does not exist" % (self.box, self.version, self.provider))
+
+        cmd = "vagrant box update --box %s --provider %s" % (self.box, self.provider)
+        rc, out, err = self.module.run_command(cmd)
+
+        if "is running the latest version" in out:
+            self.module.exit_json(changed=False, rc=rc, stdout=out, stderr=err)
+        elif rc == 0:
+            self.module.exit_json(changed=True, rc=rc, stdout=out, stderr=err)
+        else:
+            self.module.fail_json(msg=err)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            box=dict(required=True, aliases=['path', 'url'], default=None),
+            state=dict(required=False, choices=['present', 'absent', 'updated'], default="present"),
```

Comment:
required is by-default is 'False', so no need to add explicitly.

Reply:
Done.